data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateForkTest.java

```diff
@@ -14,50 +14,56 @@
 package tech.pegasys.teku.beaconrestapi.handlers.v1.beacon;
 
 import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
-import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.Mockito.when;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_REQUEST;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_INTERNAL_SERVER_ERROR;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_NOT_FOUND;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_OK;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
-import java.util.Optional;
+import java.util.concurrent.ExecutionException;
 import javax.servlet.http.HttpServletResponse;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import tech.pegasys.teku.beaconrestapi.AbstractMigratedBeaconHandlerTest;
-import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.beaconrestapi.AbstractMigratedBeaconHandlerWithChainDataProviderTest;
 import tech.pegasys.teku.infrastructure.http.HttpErrorResponse;
 import tech.pegasys.teku.infrastructure.restapi.StubRestApiRequest;
+import tech.pegasys.teku.spec.SpecMilestone;
 import tech.pegasys.teku.spec.datastructures.metadata.StateAndMetaData;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
 
-public class GetStateForkTest extends AbstractMigratedBeaconHandlerTest {
-  private final GetStateFork handler = new GetStateFork(chainDataProvider);
+public class GetStateForkTest extends AbstractMigratedBeaconHandlerWithChainDataProviderTest {
+  private GetStateFork handler;
+  private StateAndMetaData responseData;
   private final StubRestApiRequest request =
       StubRestApiRequest.builder().pathParameter("state_id", "head").build();
-  private final StateAndMetaData stateAndMetaData =
-      new StateAndMetaData(
-          dataStructureUtil.randomBeaconState(),
-          spec.getGenesisSpec().getMilestone(),
-          false,
-          false,
-          true);
```

Comment:
maybe move this to the 200 test, it was showing better info than the new changes...